whereby/browser-sdk
# Set up e2e tests

* Add a sample react app that can be used to test the Browser SDK
* Extend `useLocalMedia` to accept `MediaStreams`
* Add some e2e tests using Playwright, run them on GitHub Actions

## Test plan

1. Test the tests: follow the "Running the tests locally" guide in the `test/README.md` doc
2. Test the fake media stream story:
    1. Start storybook: `STORYBOOK_ROOM=<prod-unlocked-room-url> yarn dev`
    2. Open the "Custom UI / Local Media With Fake Media Stream" story
    3. Click the add fake audio track button
    4. Click the connect to room button
    5. Confirm you see the fake webcam and hear the JS-generated noise in your prod room.
3. See the new "Playwright Tests" CI job. It should be green.

## Gotchas

* More tests will be added in upcoming PR(s) for knocking / chat messages / (fake) cam/mic toggles, etc.
* Tests are running on macOS runners, as ubuntu's webkit browser does not fully support webRTC
* Some caching should be added to the CI flow (another upcoming task)
